### PR TITLE
feat: remove dependency on Workspace API

### PR DESCRIPTION
**Pull request by olexii4**
Signed-off-by: Oleksii Orel <oorel@redhat.com>

### What does this PR do?
Remove dependency on Workspace API.

### What issues does this PR fix or reference?
fixes https://github.com/eclipse/che/issues/21846

You can test this PR at https://eclipse-che.apps.cluster-pd8pf.pd8pf.sandbox1822.opentlc.com/dashboard/

### CODE REVIEW

File: packages/dashboard-frontend/src/store/ServerConfig/index.ts [hunk 1]
```diff
@@ -86,8 +86,7 @@ const unloadedState: State = {
     timeouts: {
       inactivityTimeout: -1,
       runTimeout: -1,
-      startTimeout: 300,
-    },
```
Comment (akurinnoy) [inline packages/dashboard-frontend/src/store/ServerConfig/index.ts:90]: I'm not sure you need to remove this timeout; it was added as part of this [PR](https://github.com/eclipse-che/che-dashboard/pull/694). However, please check it and revert this change if necessary.

File: packages/dashboard-frontend/src/typings/che.d.ts [hunk 2]
```diff
@@ -11,23 +11,6 @@
...
-    temporary?: boolean;
-    status: string;
-    namespace?: string;
-    attributes?: WorkspaceAttributes;
-    devfile: WorkspaceDevfile;
-    runtime?: WorkspaceRuntime;
-    isLocked?: boolean;
-    usedResources?: string;
-  }
-
```
Comment (akurinnoy) [inline packages/dashboard-frontend/src/typings/che.d.ts:30]: Please remove all interfaces related to Che7 workspaces.

File: packages/dashboard-frontend/src/overrides.css [hunk 3]
```diff
@@ -23,6 +23,6 @@
   padding: 0 !important;
 }
 
-span[class*="label-required"] {
+span[class*='label-required'] {
```
Comment (akurinnoy) [inline packages/dashboard-frontend/src/overrides.css:26]: Stylelinter expects double quotes here; please revert.

Comment (olexii4) [reply packages/dashboard-frontend/src/overrides.css:26]: These changes are a result of running the command
```yarn format:fix``` in the root of che-dashboard.
WDYT?

Comment (akurinnoy) [reply packages/dashboard-frontend/src/overrides.css:26]: That's weird. Let me check it.

File: packages/dashboard-frontend/src/Layout/Navigation/MainList.tsx [hunk 4]
```diff
@@ -31,7 +31,7 @@ export class NavigationMainList extends React.PureComponent<Props> {
       if (workspace.isDevWorkspace) {
         return true;
       }
-      return (workspace.ref as che.Workspace).attributes?.convertedId === undefined;
+      return (workspace.ref as any)?.attributes?.convertedId === undefined;
```
[the remark this answers is not in the document]
Comment (olexii4) [reply packages/dashboard-frontend/src/Layout/Navigation/MainList.tsx]: I have fixed it.

File: packages/dashboard-frontend/src/components/EditorTools/index.tsx [hunk 5]
```diff
@@ -24,7 +24,7 @@ import { helpers } from '@eclipse-che/common';
 import { ToggleBarsContext } from '../../contexts/ToggleBars';
 
 type Props = {
-  content: che.Workspace | devfileApi.DevWorkspace | che.WorkspaceDevfile | devfileApi.Devfile;
+  content: devfileApi.DevWorkspace | che.WorkspaceDevfile | devfileApi.Devfile;
```
[the remark this answers is not in the document]
Comment (olexii4) [reply packages/dashboard-frontend/src/components/EditorTools/index.tsx]: I have fixed it.

File: packages/dashboard-frontend/src/components/EditorTools/index.tsx [hunk 6]
```diff
@@ -113,12 +113,10 @@ class EditorTools extends React.PureComponent<Props, State> {
   }
 
   private getName(
-    content: che.Workspace | devfileApi.DevWorkspace | che.WorkspaceDevfile | devfileApi.Devfile,
+    content: devfileApi.DevWorkspace | che.WorkspaceDevfile | devfileApi.Devfile,
```
[the remark this answers is not in the document]
Comment (olexii4) [reply packages/dashboard-frontend/src/components/EditorTools/index.tsx]: I have fixed it.

File: packages/dashboard-frontend/src/components/EditorTools/index.tsx [hunk 7]
```diff
@@ -128,14 +126,11 @@ class EditorTools extends React.PureComponent<Props, State> {
   }
 
   private isWorkspace(
-    content: che.Workspace | devfileApi.DevWorkspace | che.WorkspaceDevfile | devfileApi.Devfile,
+    content: devfileApi.DevWorkspace | che.WorkspaceDevfile | devfileApi.Devfile,
```
[the remark this answers is not in the document]
Comment (olexii4) [reply packages/dashboard-frontend/src/components/EditorTools/index.tsx]: I have fixed it.

File: packages/dashboard-frontend/src/store/DockerConfig/index.ts [hunk 8]
```diff
@@ -12,11 +12,9 @@
 
 import { Action } from 'redux';
 import { AppThunk } from '..';
-import * as CheWorkspacesStore from './che';
 import * as DevWorkspacesStore from './dw';
```
[the remark this answers is not in the document]
Comment (olexii4) [reply packages/dashboard-frontend/src/store/DockerConfig/index.ts]: I have fixed it.

File: packages/dashboard-frontend/src/store/Workspaces/Settings/selectors.ts [hunk 9]
```diff
@@ -12,20 +12,12 @@
...
-export const selectAvailableStorageTypes = createSelector(selectWorkspacesSettings, settings =>
-  storageTypesService.getAvailable(settings),
-);
-
 export const selectWorkspacesSettingsError = createSelector(selectState, state => state.error);
 
-export const selectDevworkspacesEnabled = createSelector(selectState, state =>
-  isDevworkspacesEnabled(state?.settings || {}),
-);
+export const selectDevworkspacesEnabled = true;
```
[the remark this answers is not in the document]
Comment (olexii4) [reply packages/dashboard-frontend/src/store/Workspaces/Settings/selectors.ts]: I have fixed it.

File: packages/dashboard-frontend/src/store/ServerConfig/index.ts (hunk 1 above)
Comment (olexii4) [reply packages/dashboard-frontend/src/store/ServerConfig/index.ts]: I have fixed it.

File: packages/dashboard-frontend/src/typings/che.d.ts (hunk 2 above)
Comment (olexii4) [reply packages/dashboard-frontend/src/typings/che.d.ts:30]: I have fixed it.

File: packages/dashboard-frontend/src/pages/GetStarted/GetStartedTab/ImportFromGit/index.tsx [hunk 10]
```diff
@@ -60,38 +57,9 @@ export class ImportFromGit extends React.PureComponent<Props, State> {
...
-      this.devfileLocationRef.current?.invalidateInput();
-      this.showAlert({
-        key: 'load-devfile-resolver-failed',
-        title: common.helpers.errors.getMessage(e),
-        variant: AlertVariant.danger,
-      });
-    }
-  }
-
-  private showAlert(alert: AlertItem): void {
```
Comment (akurinnoy) [inline packages/dashboard-frontend/src/pages/GetStarted/GetStartedTab/ImportFromGit/index.tsx:92]: Then you need to get rid of the `removeAlerts` method and `alerts` field in the component state. Also, `factoryResolver` store is connected but not used in the component.

File: packages/dashboard-frontend/src/pages/GetStarted/__tests__/GetStarted.spec.tsx [hunk 11]
```diff
@@ -94,6 +91,7 @@ describe('Quick Add page', () => {
     await waitFor(() => expect(screen.getByRole('button', { name: 'Dummy Devfile' })).toBeTruthy());
 
     const devfileButton = screen.getByRole('button', { name: 'Dummy Devfile' });
+    await waitFor(() => expect(devfileButton).toBeTruthy());
```
Comment (akurinnoy) [inline packages/dashboard-frontend/src/pages/GetStarted/__tests__/GetStarted.spec.tsx:94]: `getBy*` throws if no elements match.

```suggestion
    const devfileButton = await screen.findByRole('button', { name: 'Dummy Devfile' });
```

File: packages/dashboard-frontend/src/pages/GetStarted/index.tsx [hunk 12]
```diff
@@ -132,7 +130,7 @@ export class GetStarted extends React.PureComponent<Props, State> {
     let workspace: Workspace | undefined;
     try {
       await this.props.createWorkspaceFromDevfile(
-        devfile,
+        devfile as any,
```
[the remark this answers is not in the document]
Comment (akurinnoy) [reply packages/dashboard-frontend/src/pages/GetStarted/index.tsx:134]:
```suggestion
        devfile,
```

File: packages/dashboard-frontend/src/pages/WorkspaceDetails/index.tsx [hunk 13]
```diff
@@ -32,10 +32,9 @@ import OverviewTab, { OverviewTab as Overview } from './OverviewTab';
 import DevfileEditorTab, { DevfileEditorTab as Editor } from './DevfileEditorTab';
 import DevworkspaceEditorTab from './DevworkspaceEditorTab';
 import { History, UnregisterCallback, Location } from 'history';
-import { isCheWorkspace, Workspace } from '../../services/workspace-adapter';
+import { Workspace } from '../../services/workspace-adapter';
 import UnsavedChangesModal from '../../components/UnsavedChangesModal';
 import WorkspaceConversionButton from './ConversionButton';
```
Comment (akurinnoy) [inline packages/dashboard-frontend/src/pages/WorkspaceDetails/index.tsx:37]: A few unused imports left here.

File: packages/dashboard-frontend/src/pages/WorkspaceDetails/index.tsx [hunk 14]
```diff
@@ -223,9 +209,6 @@ export class WorkspaceDetails extends React.PureComponent<Props, State> {
     const workspaceName = workspace.name;
     const { inlineAlertConversionError, showInlineAlertRestartWarning } = this.state;
```
Comment (akurinnoy) [inline packages/dashboard-frontend/src/pages/WorkspaceDetails/index.tsx:211]: Variables are assigned values but never used.

```suggestion
```

File: packages/dashboard-frontend/src/store/DockerConfig/index.ts [hunk 15]
```diff
@@ -10,41 +10,178 @@
...
+          registries,
+          resourceVersion,
+        });
+      } catch (e) {
+        const errorMessage = helpers.errors.getMessage(e);
+        dispatch({
+          type: 'RECEIVE_DEVWORKSPACE_CREDENTIALS_ERROR',
+          error: errorMessage,
+        });
+        throw errorMessage;
```
Comment (akurinnoy) [inline packages/dashboard-frontend/src/store/DockerConfig/index.ts:66]:
```suggestion
        throw e;
```

File: packages/dashboard-frontend/src/store/DockerConfig/selectors.ts [hunk 16]
```diff
@@ -12,23 +12,13 @@
...
   selectDwDockerConfigState,
-  (devworkspaceEnabled, cheDockerConfigState, dwDockerConfigState) => {
-    if (devworkspaceEnabled) {
-      return dwDockerConfigState;
-    } else {
-      return cheDockerConfigState;
-    }
-  },
+  dwDockerConfigState => dwDockerConfigState,
 );
```
Comment (akurinnoy) [inline packages/dashboard-frontend/src/store/DockerConfig/selectors.ts:22]:
```suggestion
const selectState = = (state: AppState) => state.dockerConfig;
```

File: packages/dashboard-frontend/src/store/ServerConfig/index.ts [hunk 17]
```diff
@@ -85,9 +85,8 @@ const unloadedState: State = {
     },
     timeouts: {
       inactivityTimeout: -1,
-      runTimeout: -1,
       startTimeout: 300,
-    },
+    } as any,
```
Comment (akurinnoy) [inline packages/dashboard-frontend/src/store/ServerConfig/index.ts:89]: Can you elaborate on why do you need to remove the `runTimeout` field?

File: packages/dashboard-frontend/src/store/Workspaces/index.ts [hunk 18]
```diff
@@ -137,7 +135,6 @@ export const actionCreators: ActionCreators = {
       try {
         const promises: Promise<unknown>[] = [
           dispatch(DevWorkspacesStore.actionCreators.requestWorkspaces()),
-          dispatch(CheWorkspacesStore.actionCreators.requestWorkspaces()),
         ];
 
         await Promise.allSettled(promises);
```
Comment (akurinnoy) [inline packages/dashboard-frontend/src/store/Workspaces/index.ts:140]:
```suggestion
        await dispatch(DevWorkspacesStore.actionCreators.requestWorkspaces());
```

File: packages/dashboard-frontend/src/store/Workspaces/index.ts [hunk 19]
```diff
@@ -301,7 +248,7 @@ export const actionCreators: ActionCreators = {
 
   createWorkspaceFromDevfile:
     (
-      devfile: che.WorkspaceDevfile | devfileApi.Devfile,
+      devfile: devfileApi.Devfile,
       namespace: string | undefined,
       infrastructureNamespace: string | undefined,
```
Comment (akurinnoy) [inline packages/dashboard-frontend/src/store/Workspaces/index.ts:253]: `namespace` and `infrastructureNamespace` are not used anymore and can be removed.

File: packages/dashboard-frontend/src/store/__mocks__/storeBuilder.ts [hunk 20]
```diff
@@ -158,29 +144,9 @@ export class FakeStoreBuilder {
...
-    return this;
-  }
-
-  public withUserPreferences(
-    preferences: che.UserPreferences,
-    isLoading = false,
-  ): FakeStoreBuilder {
-    this.state.userPreferences.preferences = preferences;
-    this.state.userPreferences.isLoading = isLoading;
+    this.state.dockerConfig.registries = registries;
```
Comment (akurinnoy) [inline packages/dashboard-frontend/src/store/__mocks__/storeBuilder.ts:147]: Please rename this method: `withDwDockerConfig` -> `withDockerConfig`

File: packages/dashboard-frontend/src/pages/GetStarted/GetStartedTab/ImportFromGit/index.tsx (hunk 10 above)
Comment (olexii4) [reply packages/dashboard-frontend/src/pages/GetStarted/GetStartedTab/ImportFromGit/index.tsx:92]: I have fixed it.

File: packages/dashboard-frontend/src/pages/GetStarted/__tests__/GetStarted.spec.tsx (hunk 11 above)
Comment (olexii4) [reply packages/dashboard-frontend/src/pages/GetStarted/__tests__/GetStarted.spec.tsx]: I have fixed it.

File: packages/dashboard-frontend/src/pages/WorkspaceDetails/index.tsx (hunk 14 above)
Comment (olexii4) [reply packages/dashboard-frontend/src/pages/WorkspaceDetails/index.tsx:210]: I have fixed it.

File: packages/dashboard-frontend/src/store/DockerConfig/index.ts (hunk 15 above)
Comment (olexii4) [reply packages/dashboard-frontend/src/store/DockerConfig/index.ts]: I have fixed it.

File: packages/dashboard-frontend/src/store/DockerConfig/index.ts [hunk 21]
```diff
@@ -10,41 +10,178 @@
...
```
[the remark this answers is not in the document]
Comment (olexii4) [reply packages/dashboard-frontend/src/store/DockerConfig/index.ts]: I have fixed it.

File: packages/dashboard-frontend/src/store/DockerConfig/selectors.ts (hunk 16 above)
Comment (olexii4) [reply packages/dashboard-frontend/src/store/DockerConfig/selectors.ts]: I have fixed it.

File: packages/dashboard-frontend/src/store/Workspaces/index.ts (hunk 18 above)
Comment (olexii4) [reply packages/dashboard-frontend/src/store/Workspaces/index.ts]: I have fixed it.

File: packages/dashboard-frontend/src/store/__mocks__/storeBuilder.ts (hunk 20 above)
Comment (olexii4) [reply packages/dashboard-frontend/src/store/__mocks__/storeBuilder.ts:147]: I have fixed it.

File: packages/dashboard-frontend/src/pages/GetStarted/index.tsx [hunk 22]
```diff
@@ -131,13 +130,7 @@ export class GetStarted extends React.PureComponent<Props, State> {
...
     let workspace: Workspace | undefined;
     try {
-      await this.props.createWorkspaceFromDevfile(
-        devfile,
-        undefined,
-        namespace,
-        attr,
-        optionalFilesContent,
-      );
+      await this.props.createWorkspaceFromDevfile(devfile as any, attr, optionalFilesContent);
```
Comment (akurinnoy) [inline packages/dashboard-frontend/src/pages/GetStarted/index.tsx:133]:
```suggestion
      await this.props.createWorkspaceFromDevfile(devfile, attr, optionalFilesContent);
```

Comment (olexii4) [reply packages/dashboard-frontend/src/pages/GetStarted/index.tsx]: I have fixed it.

File: packages/dashboard-frontend/src/pages/GetStarted/GetStartedTab/SamplesListGallery.tsx [hunk 23]
```diff
@@ -218,9 +215,9 @@ export class SamplesListGallery extends React.PureComponent<Props, State> {
       .sort(SamplesListGallery.sortByDisplayName)
       .sort(SamplesListGallery.sortByVisibleTag)
       .sort(SamplesListGallery.sortByEmptyWorkspaceTag)
-      .map(meta => (
+      .map((meta: che.DevfileMetaData, index: number) => (
         <SampleCard
-          key={meta.links.self}
+          key={`${meta.links.self}_${index}`}
```
Comment (akurinnoy) [inline packages/dashboard-frontend/src/pages/GetStarted/GetStartedTab/SamplesListGallery.tsx:220]:
```suggestion
          key={meta.links.self}
```

File: packages/dashboard-frontend/src/services/helpers/devworkspace.ts [hunk 24]
```diff
@@ -10,17 +10,13 @@
...
+import * as devfileApi from '../devfileApi/devfileApi';
+
 /**
  * Check to see if the workspace is a web terminal
  * @param workspaceOrDevfile The workspace or devfile you want to check
  */
-export function isWebTerminal(
-  workspaceOrDevfile: che.Workspace | api.che.workspace.devfile.Devfile,
-): boolean {
+export function isWebTerminal(workspaceOrDevfile: devfileApi.DevWorkspace): boolean {
```
Comment (akurinnoy) [inline packages/dashboard-frontend/src/services/helpers/devworkspace.ts:19]:
```suggestion
export function isWebTerminal(workspace: devfileApi.DevWorkspace): boolean {
```

File: packages/dashboard-frontend/src/pages/GetStarted/GetStartedTab/SamplesListGallery.tsx (hunk 23 above)
Comment (olexii4) [reply packages/dashboard-frontend/src/pages/GetStarted/GetStartedTab/SamplesListGallery.tsx]: Done

File: packages/dashboard-frontend/src/services/helpers/devworkspace.ts (hunk 24 above)
Comment (olexii4) [reply packages/dashboard-frontend/src/services/helpers/devworkspace.ts]: Done

File: packages/dashboard-backend/src/routes/api/devfileSchema.ts [hunk 25]
```diff
@@ -15,7 +15,6 @@ import { baseApiPath } from '../../constants/config';
 import { getSchema } from '../../services/helpers';
 import { devfileVersionSchema } from '../../constants/schemas';
 import { restParams } from '../../typings/models';
-import * as devfileSchemaV100 from '../../devfileSchemas/1.0.0/devfile.json';
```
Comment (ibuziuk) [inline packages/dashboard-backend/src/routes/api/devfileSchema.ts:18]: @olexii4 since we are not doing the validation of v1 I'm wondering if the factory automatic conversion from v1 to v2 is going to still work?

**Review by ibuziuk — APPROVED** [review]

File: packages/dashboard-frontend/src/services/helpers/devworkspace.ts (hunk 24 above)
Comment (olexii4) [reply packages/dashboard-frontend/src/services/helpers/devworkspace.ts]: Done

File: packages/dashboard-backend/src/routes/api/devfileSchema.ts (hunk 25 above)
Comment (olexii4) [reply packages/dashboard-backend/src/routes/api/devfileSchema.ts:18]: All validation for the Devfile v1 during factory flow on the che-server side.
Знімок екрана 2023-01-11 о 12 17 47

**Review by akurinnoy — APPROVED** [review]
